File: challenges/medium/02_inventory/logger.py
class RequestLog:
    log = []  # store request IDs for this service

    def __init__(self, service_name):
        self.service_name = service_name

    def record(self, request_id):
        self.log.append(request_id)

    def get_log(self):
        return list(self.log)

File: challenges/medium/08_inventory_js/logger.js
const log = [];    // bug: module-level array, shared across every RequestLog instance

class RequestLog {
  constructor(serviceName) {
    this.serviceName = serviceName;
  }

  record(requestId) {
    log.push(requestId);
  }

  getLog() {
    return [...log];
  }
}

module.exports = { RequestLog };
